redhat-developer/devconsole-operator
# Add gitsource CR reference in component CR

This PR makes it possible to support importing private repo (By referring `GitSource` CR to Component `codebase`)

### How to test
1. Create `Secret` object with username and password.
2. Create `GitSource` object which refers to `Secret` name
3. Create `Component` object where `codebase` field should be name of `GitSource` CR.
4. You should see `BuildConfig` and `ImageStream` generated.
5. `BuildConfig` should have git url specified in the `GitSource` and build should be passed. 

- [x]  Need to do one round of testing
- [x]  Unit tests need to be written.
- [x]  Rename `codebase` field to `gitSourceRef`
